packages/cli/src/commands/account/new.ts

```diff
@@ -1,52 +1,61 @@
-import { ec as EC } from 'elliptic'
-import Web3 from 'web3'
+import { generateKeys, generateMnemonic, MnemonicLanguages } from '@celo/utils/lib/account'
+import { privateKeyToAddress } from '@celo/utils/lib/address'
+import { flags } from '@oclif/command'
 import { LocalCommand } from '../../base'
 import { printValueMap } from '../../utils/cli'
-import { ReactNativeBip39MnemonicGenerator } from '../../utils/key_generator'
 
 export default class NewAccount extends LocalCommand {
   static description =
-    'Creates a new account locally and print out the key information. Save this information for local transaction signing or import into a Celo node.'
+    "Creates a new account locally using the Celo Derivation Path (m/44'/52752'/0/0/indexAddress) and print out the key information. Save this information for local transaction signing or import into a Celo node. Ledger: this command has been tested swapping mnemonics with the Ledger successfully (only supports english)"
 
   static flags = {
     ...LocalCommand.flags,
+    password: flags.string({
+      description: 'Choose a password to generate the keys',
+    }),
+    indexAddress: flags.integer({
+      default: 0,
+      description: 'Choose the index address of the derivation path',
+    }),
+    language: flags.string({
+      options: [
+        'chinese_simplified',
+        'chinese_traditional',
+        'english',
+        'french',
+        'italian',
+        'japanese',
+        'korean',
+        'spanish',
+      ],
+      default: 'english',
+      description:
+        "Language for the mnemonic words. **WARNING**, some hardware wallets don't support other languages",
+    }),
   }
 
-  static examples = ['new']
+  static examples = [
+    'new',
+    'new --password 12341234',
+    'new --language spanish',
+    'new --password 12341234 --language japanese --indexAddress 5',
+  ]
 
-  static getRandomMnemonic(): string {
-    return ReactNativeBip39MnemonicGenerator.generateMnemonic()
-  }
-
-  // Do same as what the mobile wallet app does
-  // https://github.com/celo-org/celo-monorepo/blob/16f01a14ef0b0a15c0ffe8bb7b940670a0654ad3/packages/mobile/src/import/saga.ts
-  static getPrivateKey(mnemonic: string): string {
-    return ReactNativeBip39MnemonicGenerator.mnemonicToSeedHex(mnemonic)
-  }
-
-  static getPublicKey(privateKey: string): string {
-    const ec = new EC('secp256k1')
-    const ecKeyPair: EC.KeyPair = ec.keyFromPrivate(Buffer.from(privateKey, 'hex'), 'hex')
-    const ecPublicKey: string = ecKeyPair.getPublic('hex')
-    return ecPublicKey
-  }
-
-  static generateAccountAddressFromPrivateKey(privateKey: string): string {
-    if (!privateKey.startsWith('0x')) {
-      privateKey = '0x' + privateKey
+  static languageOptions(language: string): MnemonicLanguages | undefined {
+    if (language) {
+      return (MnemonicLanguages[language as any] as unknown) as MnemonicLanguages
```

Comment:
Can this type casting be cleaned up? It seems like a ts-ignore may be better than 3x casts if necessary

Reply:
Haha, you are right